pkg/aggregator/aggregator.go

```diff
@@ -761,19 +676,18 @@ func (agg *BufferedAggregator) run() {
 		case <-agg.stopChan:
 			log.Info("Stopping aggregator")
 			return
-		case <-agg.health.C:
-		case <-agg.TickerChan:
-			start := time.Now()
-			agg.Flush(start, false)
+		case trigger := <-agg.flushChan:
+			agg.Flush(trigger)
 
 			// Do this here, rather than in the Flush():
 			// - make sure Shrink doesn't happen concurrently with sample processing.
 			// - we don't need to Shrink() on stop
 			agg.tagsStore.Shrink()
 
-			addFlushTime("MainFlushTime", int64(time.Since(start)))
+			addFlushTime("MainFlushTime", int64(time.Since(trigger.time)))
```

Comment:
The duration is not the same as before as `trigger.time` is set to the current time before putting in the channel.